javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java

```diff
@@ -24,51 +24,65 @@
 import com.github.javaparser.Position;
 import com.github.javaparser.utils.Utils;
 
-public class SourcePrinter {
+import java.util.Deque;
+import java.util.LinkedList;
 
-    private final String indentation;
-    private final int indentationLength;
+public class SourcePrinter {
     private final String endOfLineCharacter;
-    private int level = 0;
-    private boolean indented = false;
+    private final String indentation;
+
+    private final Deque<String> indents = new LinkedList<>();
     private final StringBuilder buf = new StringBuilder();
     private Position cursor = new Position(1, 0);
+    private boolean indented=false;
 
     SourcePrinter(final String indentation, final String endOfLineCharacter) {
         this.indentation = indentation;
-        this.indentationLength = indentation.length();
         this.endOfLineCharacter = endOfLineCharacter;
+        indents.push("");
     }
 
     public SourcePrinter indent() {
-        level++;
+        String currentIndent = indents.peek();
+        indents.push(currentIndent + indentation);
+        return this;
+    }
+
+    public SourcePrinter indentTo(int column) {
+        StringBuilder newIndent = new StringBuilder(indents.peek());
+        while (newIndent.length() < column) {
+            newIndent.append(' ');
+        }
+        indents.push(newIndent.toString());
         return this;
     }
 
     public SourcePrinter unindent() {
-        level--;
+        indents.pop();
         return this;
     }
 
+    private void append(String arg) {
```

Comment:
Should also be used in `print()` and `println()`

Reply:
Yes, except println because that does funny cursor logic.

Reply:
Ah, right, didn't look closely enough, then only for `print`